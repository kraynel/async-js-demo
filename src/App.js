// @flow

import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

import { Provider } from 'react-redux';

import type StoreType from './modules/reducers';
import createStore from './modules/store';

import CallbackCall from './asyncCalls/CallbackCall';
import PromiseCall from './asyncCalls/PromiseCall';
import AsyncAwaitCall from './asyncCalls/AsyncAwaitCall';
import SagaCall from './asyncCalls/SagaCall';

type StateType = {
  store: ?StoreType
};

class App extends Component<null, StateType> {
  state = {
    store: null
  };

  componentDidMount() {
    const store = createStore();
    this.setState({ store });
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <CallbackCall />
        <PromiseCall />
        <AsyncAwaitCall />

        {this.state.store && (
          <Provider store={this.state.store}>
            <SagaCall />
          </Provider>
        )}
      </div>
    );
  }
}

export default App;
