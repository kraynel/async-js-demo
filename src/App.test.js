import React from 'react';
import renderer from 'react-test-renderer';

import App from './App';

describe('App', () => {
  it('should render without crashing', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
