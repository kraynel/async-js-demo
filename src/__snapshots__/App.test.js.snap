// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render without crashing 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-image"
      src="asynchronous-javascript.png"
    />
  </header>
  <div>
    <button
      onClick={[Function]}
    >
      Callback Call
    </button>
    <span>
      In real world
    </span>
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Promise Call
    </button>
    <span>
      In real world
    </span>
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Async/await Call
    </button>
    <span>
      In real world
    </span>
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Saga Call
    </button>
    <span>
      In real world
    </span>
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Observable Call
    </button>
    <span>
      In real world
    </span>
  </div>
</div>
`;
